patricia/patricia.go

```diff
@@ -19,6 +19,29 @@ type (
 	VisitorFunc func(prefix Prefix, item Item) error
 )
 
+const (
+	DefaultMaxPrefixPerNode         = 10
+	DefaultMaxChildrenPerSparseNode = 8
+)
+
+type Option struct {
+	// Max prefix length that is kept in a single trie node.
+	MaxPrefixPerNode int
+	// Max children to keep in a node in the sparse mode.
+	MaxChildrenPerSparseNode int
+}
+
+func NewOption(maxPrefixPerNode int, maxChildrenPerSparseNode int) *Option {
```

Comment:
Sorry for the late response. We don't need to sort every time when adding something. Just sort it before working like this(https://github.com/ishidawataru/go-patricia/commit/04d67b07340f7bab5a4b649ddc7f95fd9c851043). I think patricia tree should support waking in order of prefix. I checked py-radix, and it works like this.